### ISSUE TITLE
Consumer stuck out of permits

### ISSUE THREAD

**Issue by alanhoff** [streamnative/pulsar-rs#114]
Hello there,

I have this example code below, it seems like the Consumer stops receiving messages after it's out of permits when I set `batch_size: Some(u32)` in the Producer. If I remove the `batch_size` from the Producer then everything works as expected and the flow control will refill the Consumer's permits.

```rust
use futures::TryStreamExt;
use pulsar::message::proto::command_subscribe::SubType;
use pulsar::{producer, Consumer, Pulsar, TokioExecutor};
use std::sync::atomic::{AtomicUsize, Ordering};
use std::sync::Arc;

#[tokio::main]
async fn main() {
    env_logger::init();

    let addr = "pulsar://127.0.0.1:6650";
    let pulsar: Pulsar<_> = Pulsar::builder(addr, TokioExecutor).build().await.unwrap();
    let produced = Arc::new(AtomicUsize::new(0));
    let consumed = Arc::new(AtomicUsize::new(0));

    // Producer
    let producer = {
        let pulsar = pulsar.clone();
        let produced = produced.clone();
        tokio::spawn(async move {
            let mut producer = pulsar
                .producer()
                .with_topic("persistent://public/default/my-topic")
                .with_name("my producer")
                .with_options(producer::ProducerOptions {
                    batch_size: Some(5),
                    ..Default::default()
                })
                .build()
                .await
                .unwrap();

            loop {
                producer.send(b"hello world".to_vec()).await.unwrap();
                produced.fetch_add(1, Ordering::SeqCst);
            }
        })
    };

    // Consumer
    let consumer = {
        let pulsar = pulsar.clone();
        let consumed = consumed.clone();
        tokio::spawn(async move {
            let mut consumer: Consumer<Vec<u8>, _> = pulsar
                .consumer()
                .with_topic("persistent://public/default/my-topic")
                .with_subscription_type(SubType::Exclusive)
                .build()
                .await
                .unwrap();

            loop {
                let msg = consumer.try_next().await.unwrap().unwrap();
                consumer.ack(&msg).await.unwrap();
                consumed.fetch_add(1, Ordering::SeqCst);
            }
        })
    };

    let stats = {
        let produced = produced.clone();
        let consumed = consumed.clone();

        tokio::spawn(async move {
            loop {
                tokio::time::delay_for(std::time::Duration::from_millis(5000)).await;
                let produced = produced.swap(0, Ordering::SeqCst);
                let consumed = consumed.swap(0, Ordering::SeqCst);

                println!("Produced {} msgs/sec", produced / 5);
                println!("Consumed {} msgs/sec", consumed / 5);
            }
        })
    };

    tokio::join!(producer, consumer, stats);
}
```

**Comment by Geal** [comment]
nice catch. It should be fixed by d468a4e, can you test on your side?

**Comment by alanhoff** [comment]
Works as expected! Thx for the quick fix! 🚀
